### ISSUE TITLE
Investigate Legacy Hearings with No Associations

### ISSUE THREAD

**Issue by lomaxap** [department-of-veterans-affairs/caseflow#10643]
```ruby
vacols_hearings = VACOLS::CaseHearing.joins(:brieff).where("hearing_date >= '2019-04-01'").where(hearing_type: ["C", "V"]).where("brieff.bfcurloc = 'CASEFLOW'")

legacy_hearings = LegacyHearing.where(vacols_id: vacols_hearings.pluck(:hearing_pkseq))
associations = HearingTaskAssociation.where(hearing: legacy_hearings)

no_associations = legacy_hearings.pluck(:id) - associations.pluck(:hearing_id)

legacy_hearings_with_no_associations = LegacyHearing.where(id: no_associations)
appeals_with_no_associations = LegacyAppeal.where(id: legacy_hearings_with_no_associations.pluck(:appeal_id))

```
reveals 35 legacy hearings with no associations (data commented below)

---
if
```ruby
hearings_for_this_appeal == total_hearing_tasks_for_this_appeal &&
open_hearing_tasks_for_this_appeal_with_no_associations == 
number_hearings_with_no_associations_for_this_appeal == 1
```
we may be able to safely associate the hearing with the unassociated hearing task
 
---
if 
```ruby
hearings_for_this_appeal > total_hearing_tasks_for_this_appeal
```
We may need to create new hearing task trees. We'll also need some way to determine which trees are stale and which should be the latest state of the appeal

---
This still doesn't answer the question around how these hearing weren't associated in the first place. I don't see any patterns around `scheduled_for` date. We can't determine when these hearings were scheduled, but these do not appear to be hearings that were missed in the backfill. My current thought is that these were all scheduled around the same time when the hearing task association logic was not completely implemented, and they were lost in the shuffle. 

May want to consider creating a Looker alert to monitor hearings that are created but not given an association.

**Comment by lomaxap** [comment]
```ruby
{48222=>
  [{:id=>84292,
    :appeal_id=>48222,
    :scheduled_for=>Tue, 09 Apr 2019 14:30:00 EDT -04:00,
    :disposition=>"no_show",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>1}],
 510093=>
  [{:id=>84297,
    :appeal_id=>510093,
    :scheduled_for=>Tue, 09 Apr 2019 16:30:00 EDT -04:00,
    :disposition=>"cancelled",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>0}],
 607038=>
  [{:id=>85269,
    :appeal_id=>607038,
    :scheduled_for=>Mon, 08 Jul 2019 16:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>1}],
 35810=>
  [{:id=>85302,
    :appeal_id=>35810,
    :scheduled_for=>Mon, 08 Jul 2019 18:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>0}],
 24677=>
  [{:id=>85451,
    :appeal_id=>24677,
    :scheduled_for=>Thu, 04 Apr 2019 08:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>3,
    :open_hearing_tasks_for_this_appeal=>1}],
 522832=>
  [{:id=>85717,
    :appeal_id=>522832,
    :scheduled_for=>Mon, 08 Apr 2019 08:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>1}],
 149171=>
  [{:id=>85788,
    :appeal_id=>149171,
    :scheduled_for=>Wed, 14 Aug 2019 10:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>5,
    :open_hearing_tasks_for_this_appeal=>1}],
 4201=>
  [{:id=>86063,
    :appeal_id=>4201,
    :scheduled_for=>Wed, 14 Aug 2019 10:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>1}],
 571682=>
  [{:id=>86110,
    :appeal_id=>571682,
    :scheduled_for=>Wed, 10 Apr 2019 12:30:00 EDT -04:00,
    :disposition=>"held",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>1}],
 116183=>
  [{:id=>86829,
    :appeal_id=>116183,
    :scheduled_for=>Fri, 05 Apr 2019 13:30:00 EDT -04:00,
    :disposition=>"no_show",
    :hearings_for_this_appeal=>3,
    :open_hearing_tasks_for_this_appeal=>1}],
 546095=>
  [{:id=>86998,
    :appeal_id=>546095,
    :scheduled_for=>Thu, 11 Apr 2019 00:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>8,
    :open_hearing_tasks_for_this_appeal=>1},
   {:id=>87018,
    :appeal_id=>546095,
    :scheduled_for=>Fri, 05 Apr 2019 21:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>8,
    :open_hearing_tasks_for_this_appeal=>1},
   {:id=>87019,
    :appeal_id=>546095,
    :scheduled_for=>Wed, 10 Apr 2019 18:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>8,
    :open_hearing_tasks_for_this_appeal=>1},
   {:id=>87057,
    :appeal_id=>546095,
    :scheduled_for=>Fri, 12 Apr 2019 18:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>8,
    :open_hearing_tasks_for_this_appeal=>1},
   {:id=>87059,
    :appeal_id=>546095,
    :scheduled_for=>Fri, 12 Apr 2019 18:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>8,
    :open_hearing_tasks_for_this_appeal=>1},
   {:id=>87061,
    :appeal_id=>546095,
    :scheduled_for=>Thu, 11 Apr 2019 03:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>8,
    :open_hearing_tasks_for_this_appeal=>1},
   {:id=>87062,
    :appeal_id=>546095,
    :scheduled_for=>Thu, 11 Apr 2019 00:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>8,
    :open_hearing_tasks_for_this_appeal=>1}],
 546096=>
  [{:id=>87000,
    :appeal_id=>546096,
    :scheduled_for=>Thu, 11 Apr 2019 05:30:00 EDT -04:00,
    :disposition=>"held",
    :hearings_for_this_appeal=>4,
    :open_hearing_tasks_for_this_appeal=>1},
   {:id=>87022,
    :appeal_id=>546096,
    :scheduled_for=>Fri, 12 Apr 2019 14:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>4,
    :open_hearing_tasks_for_this_appeal=>1},
   {:id=>87023,
    :appeal_id=>546096,
    :scheduled_for=>Wed, 10 Apr 2019 23:30:00 EDT -04:00,
    :disposition=>"held",
    :hearings_for_this_appeal=>4,
    :open_hearing_tasks_for_this_appeal=>1}],
 383898=>
  [{:id=>87001,
    :appeal_id=>383898,
    :scheduled_for=>Thu, 11 Apr 2019 03:30:00 EDT -04:00,
    :disposition=>"held",
    :hearings_for_this_appeal=>5,
    :open_hearing_tasks_for_this_appeal=>0},
   {:id=>87031,
    :appeal_id=>383898,
    :scheduled_for=>Fri, 12 Apr 2019 11:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>5,
    :open_hearing_tasks_for_this_appeal=>0},
   {:id=>87037,
    :appeal_id=>383898,
    :scheduled_for=>Wed, 10 Apr 2019 23:30:00 EDT -04:00,
    :disposition=>"held",
    :hearings_for_this_appeal=>5,
    :open_hearing_tasks_for_this_appeal=>0}],
 594968=>
  [{:id=>87060,
    :appeal_id=>594968,
    :scheduled_for=>Mon, 01 Apr 2019 20:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>1}],
 33164=>
  [{:id=>87109,
    :appeal_id=>33164,
    :scheduled_for=>Wed, 10 Apr 2019 12:30:00 EDT -04:00,
    :disposition=>"held",
    :hearings_for_this_appeal=>3,
    :open_hearing_tasks_for_this_appeal=>1}],
 21882=>
  [{:id=>87487,
    :appeal_id=>21882,
    :scheduled_for=>Tue, 09 Apr 2019 12:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>1}],
 594860=>
  [{:id=>88260,
    :appeal_id=>594860,
    :scheduled_for=>Thu, 11 Apr 2019 10:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>3,
    :open_hearing_tasks_for_this_appeal=>1}],
 58576=>
  [{:id=>88299,
    :appeal_id=>58576,
    :scheduled_for=>Mon, 15 Jul 2019 10:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>4,
    :open_hearing_tasks_for_this_appeal=>1},
   {:id=>88302,
    :appeal_id=>58576,
    :scheduled_for=>Thu, 11 Apr 2019 11:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>4,
    :open_hearing_tasks_for_this_appeal=>1}],
 67383=>
  [{:id=>88303,
    :appeal_id=>67383,
    :scheduled_for=>Thu, 11 Apr 2019 10:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>3,
    :open_hearing_tasks_for_this_appeal=>1}],
 612724=>
  [{:id=>89984,
    :appeal_id=>612724,
    :scheduled_for=>Tue, 02 Apr 2019 12:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>1}],
 613034=>
  [{:id=>91704,
    :appeal_id=>613034,
    :scheduled_for=>Wed, 01 May 2019 14:30:00 EDT -04:00,
    :disposition=>"cancelled",
    :hearings_for_this_appeal=>3,
    :open_hearing_tasks_for_this_appeal=>1}],
 86599=>
  [{:id=>91756,
    :appeal_id=>86599,
    :scheduled_for=>Mon, 29 Apr 2019 08:30:00 EDT -04:00,
    :disposition=>"postponed",
    :hearings_for_this_appeal=>3,
    :open_hearing_tasks_for_this_appeal=>1}],
 82343=>
  [{:id=>94560,
    :appeal_id=>82343,
    :scheduled_for=>Mon, 01 Apr 2019 14:30:00 EDT -04:00,
    :disposition=>"held",
    :hearings_for_this_appeal=>2,
    :open_hearing_tasks_for_this_appeal=>1}],
 557826=>
  [{:id=>101272,
    :appeal_id=>557826,
    :scheduled_for=>Thu, 13 Jun 2019 08:30:00 EDT -04:00,
    :disposition=>nil,
    :hearings_for_this_appeal=>3,
    :open_hearing_tasks_for_this_appeal=>1}]}
```

**Comment by lomaxap** [comment]
### notes
- Combed this list and manually created associations where it made sense 
- The new cutoff for when hearings were no longer scheduled in VACOLS is on or before 04/12/2019
- There are a handful with no associations after that date that have tasks in a sensible state, so I will leave them be: `LegacyAppeal.where(vacols_id: ["3136809", "3601526", "3003035", "3126735"])`
- A few Legacy Appeals have a tasks state that I couldn't work out. Will need to follow up with Sal `LegacyAppeal.where(vacols_id: ["3295173", "3440775", "3209620"])`
- Identified bug in legacy hearings where if a user postpones on the same day, the new HearingTask will be associated with the old (postponed) hearing (resolved with #11144)
